scripts/flag_reviews.py

```diff
@@ -1,66 +1,69 @@
-    df['flag'] = None
+    df = denormalised_reviews_to_df(DATABASE_URL)
     duplicate_reviews_dict = {}
 
     for index, row in df.iterrows():
-        store_name = row['Store Name']
-        useful_info = row['Useful Information'].strip().lower() if row['Useful Information'] else ''
-        safety_info = row['Safety Observations'].strip().lower() if row['Safety Observations'] else ''
-        timestamp = pd.to_datetime(row['Timestamp'])
+        store_name = row['name']
+        useful_info = row['availabilityInfo'].strip().lower() if row['availabilityInfo'] else ''
+        safety_info = row['safetyInfo'].strip().lower() if row['safetyInfo'] else ''
+        timestamp = pd.to_datetime(row['updatedAt'])
 
-        dict_key = (store_name.strip().lower(), row['User IP'])
+        dict_key = (store_name.strip().lower(), row['ip'])
 
         if dict_key in duplicate_reviews_dict:
-            review_proximity = pd.Timedelta(timestamp - pd.to_datetime(duplicate_reviews_dict[dict_key]['Timestamp'])).seconds / 60
+            # Only count as duplicate if the two reviews are 300 mins apart
+            review_proximity = pd.Timedelta(timestamp - pd.to_datetime(duplicate_reviews_dict[dict_key]['updatedAt'])).seconds / 60
             if abs(review_proximity) < 300:
-                # print("With review proxmity ", review_proximity, " minutes")
-                if (safety_info == duplicate_reviews_dict[dict_key]['Safety Observations'].strip().lower()) and (useful_info == duplicate_reviews_dict[dict_key]['Useful Information'].strip().lower()):
-                    # print("duplicate found: ", safety_info, duplicate_reviews_dict[dict_key]['Safety Observations'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['Useful Information'].strip().lower())
-                    row['flag'] = 'Duplicate'
+                if (safety_info == duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower()) and (useful_info == duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower() and pd.notnull(row['openingTime']) and pd.notnull(row['closingTime'])):
+                    # print("duplicate found: ", safety_info, duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower())
+                    df.at[index , 'flag'] = Flag.DUPLICATE
                     continue                    
                 # else: 
-                    # print("duplicate NOT found: ", safety_info, duplicate_reviews_dict[dict_key]['Safety Observations'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['Useful Information'].strip().lower())
+                    # print("duplicate NOT found: ", safety_info, duplicate_reviews_dict[dict_key]['safetyInfo'].strip().lower(), "and useful info ", useful_info, duplicate_reviews_dict[dict_key]['availabilityInfo'].strip().lower())
```

Comment:
i prefer not having commented code as much as possible. if it's really required, ideally we should add a comment on why it has been commented